packages/react/src/primitives/View/View.tsx

```diff
@@ -21,15 +21,27 @@ const ViewPrimitive = <Element extends ElementType = 'div'>(
 ) => {
   const { propStyles, nonStyleProps } = useStyles(rest, style);
 
+  // No need to do extra work on style props when it is not rendered as a HTML element
+  // e.g. <Image as={NextImage} src="" width={300} height={300} />, just let all props pass to NextImage
+  const _rest =
```

Comment:
I think this might need a better name. How about:
```suggestion
  const stylesAndRest =
```